demos/coreMQTT/mqtt_demo_mutual_auth.c

```diff
@@ -779,6 +779,8 @@ static BaseType_t prvConnectToServerWithBackoffRetries( NetworkContext_t * pxNet
         }
     } while( ( xNetworkStatus != TRANSPORT_SOCKET_STATUS_SUCCESS ) && ( xBackoffStatus == pdPASS ) );
 
+    xStatus = ( xBackoffStatus == pdPASS ) ? pdPASS : pdFAIL;
+
     return xStatus;
```

Comment:
Although, I think we care more about `xNetworkStatus` instead
```suggestion
    return ( xNetworkStatus == TRANSPORT_SOCKET_STATUS_SUCCESS ) ? pdPASS : pdFAIL;
```